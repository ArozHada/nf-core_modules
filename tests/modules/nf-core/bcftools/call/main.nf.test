nextflow_process {

    name "Test Process BCFTOOLS_CALL"
    script "modules/nf-core/bcftools/call/main.nf"
    process "BCFTOOLS_CALL"
    config "tests/modules/nf-core/bcftools/call/nextflow.config"
    tag "bcftools_call"

    test("Sarscov2 Illumina VCF") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                            [ id:'out', single_end:false ], // meta map
                            file(params.test_data['sarscov2']['illumina']['test_vcf_gz'], checkIfExists: true),
                            file(params.test_data['sarscov2']['illumina']['test_vcf_gz_tbi'], checkIfExists: true)
                           ]
                input[1] = []
                input[2] = []
                input[3] = []           
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match("vcf") }
            )
        }
    }

    test("Sarscov2 Illumina VCF With Regions Markers") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                            [ id:'out', single_end:false ], // meta map
                            file(params.test_data['sarscov2']['illumina']['test_vcf_gz'], checkIfExists: true),
                            file(params.test_data['sarscov2']['illumina']['test_vcf_gz_tbi'], checkIfExists: true)
                           ]
                input[1] = file(params.test_data['sarscov2']['illumina']['test3_vcf_gz'], checkIfExists: true)
                input[2] = file(params.test_data['sarscov2']['illumina']['test2_vcf_targets_tsv_gz'], checkIfExists: true)
                input[3] = []           
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match("vcf_regions_markers") }
            )
        }
    }
}
