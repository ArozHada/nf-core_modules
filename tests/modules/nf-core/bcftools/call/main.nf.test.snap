{
    "vcf": {
        "content": [
            [
                "out.vcf.gz:md5,4f24467157f5c7a3b336481acf0c8a65",
                "versions.yml:md5,458be30ad798e9b063594a9762e77b71"
            ]
        ],
        "timestamp": "2023-04-08T15:12:00+0000"
    },
    "vcf_regions_markers": {
        "content": [
            [
                "out.vcf.gz:md5,ed691eee98c5c5abdf751e3006b8e206",
                "versions.yml:md5,458be30ad798e9b063594a9762e77b71"
            ]
        ],
        "timestamp": "2023-04-08T15:12:00+0000"
    }
}